core-bundle/src/Resources/contao/drivers/DC_Table.php

```diff
@@ -3207,6 +3197,8 @@ protected function submit()
 				$arrValues['ptable'] = $this->ptable;
 			}
 
+			$this->denyAccessUnlessGranted(ContaoCorePermissions::DC_PREFIX . $this->strTable, new UpdateAction($this->strTable, $currentRecord, $this->arrSubmit));
+
 			$objUpdateStmt = $this->Database
 				->prepare("UPDATE " . $this->strTable . " %s WHERE " . implode(' AND ', $this->procedure))
 				->set($arrValues)
```

Comment:
at the end of this method (I can't add a comment there), we should clean the currentRecordCache to make sure the new data is fetched when used by someone (e.g. in the `onsubmit_callback`)!

Reply:
Correct, adjusted.